controllers/usersController.js

```diff
@@ -0,0 +1,141 @@
+const bcrypt = require('bcrypt');
+const jwt = require('jsonwebtoken');
+const User = require('../models/usersModel');
+
+exports.createUser = async (req, res) => {
+  try {
+    const { email, password } = req.body;
+    const user = await User.findOne({ email });
+    if (user) res.status(409).send({ status: 'fail', err: 'The user already exists!'});
+    else {
+      const saltRounds = 10;
+      const hash = await bcrypt.hash(password, saltRounds);
+      const newUser = {
+        email,
+        password: hash,
+      };
+      const createdUser = await User.create(newUser);
+      res.status(201).send({ email: createdUser.email, id: createdUser._id }); // eslint-disable-line
+    }
+  } catch (err) {
+    res.status(500).send({ status: 'fail', err });
+  }
+};
+
+exports.login = async (req, res) => {
+  try {
+    // check if there is a basic authorization header
+    const basicAuth = req.headers.authorization.split(' ');
+    if (basicAuth.length !== 2 || basicAuth[0] !== 'Basic') {
+      throw new Error('Missing basic authorization header.');
+    }
+    // set email and password from the header and check the database for a match
+    const [email, password] = Buffer.from(basicAuth[1], 'base64').toString().split(':');
+    const user = await User.findOne({ email });
+    if (!user) res.status(401).send({ status: 'fail', err: 'The user doesn\'t exist!'});
+    else {
+      const match = await bcrypt.compare(password, user.password);
+      // if there is a match send the token to the client
+      if (match) {
+        const filteredUser = Object.keys(user._doc) // eslint-disable-line
+          .filter(key => key === 'email' || key === '_id')
+          .reduce((newObj, key) => {
+            newObj[key] = user[key]; // eslint-disable-line
+            return newObj;
+          }, {});
+        // sign and send the json web token
+        jwt.sign(filteredUser, process.env.SECRET, (err, token) => {
+          if (err) {
+            console.log(err); // eslint-disable-line
+            res.status(401).send({ status: 'fail', err: 'Error signing, please retry!' });
+          } else {
+            res.status(200).send({ id_token: token });
+          }
+        });
+      } else {
+        res.set({
+          'WWW-Authenticate': 'Basic',
+        });
+        res.status(401).send({ status: 'fail', err: 'Could not Login, the inserted password is wrong!' });
+      }
+    }
+  } catch (err) {
+    console.log(err); // eslint-disable-line
+    res.status(500).send({ status: 'fail', err });
+  }
+};
+
+exports.createLink = async (req, res) => {
```

Comment:
Personally, I would prefer to have another controller, which handles for signed-in users all the requests (create, get, etc.) with respect to their link list.